src/vss_tools/vspec/vssexporters/vss2apigear.py

```diff
@@ -0,0 +1,522 @@
+    if (SolutionLayers.QT6 in layers):
+        file.write(SOLUTION_TEMPLATE_LAYER_QT6
+                   .format(module_name=module_filename, output_path=layers[SolutionLayers.QT6]))
+
+    log.info("Solution file generated at location : " + path)
+
+
+def generate_module(file, module: ApiGearModule, module_name: str):
+    file.write(MODULE_TEMPLATE_START.format(module_name))
+    file.write("interfaces:\n")
+
+    for interface in module.interfaces:
+        if len(interface.properties) == 0:
+            continue
+
+        file.write(f"  - name: {interface.name}\n")
+        file.write("    properties:\n")
+
+        for prop_key, prop_value in interface.properties.items():
+            file.write(f"      - name: {prop_key}\n")
+            if prop_value.description is not None:
+                file.write(f"        description: \"{prop_value.description}\"\n")
+            file.write(f"        type: {prop_value.type.type}\n")
+            if prop_value.type.is_array:
+                file.write("        array: true\n")
+
+    if len(module.enumerations) > 0:
+        file.write("\n")
+        file.write("enums:\n")
+
+        for enumeration in module.enumerations:
+            file.write(f"  - name: {enumeration.name}\n")
+            file.write("    members:\n")
+
+            for variant, value in enumeration.variants:
+                file.write(f"      - name: {variant}\n")
+                if value is not None:
+                    file.write(f"        value: {value}\n")
+
+    if len(module.structures) > 0:
+        file.write("\n")
+        file.write("structs:\n")
+
+        for structure in module.structures:
+            file.write(f"  - name: {structure.name}\n")
+            file.write("    fields:\n")
+
+            for prop_key, prop_value in structure.properties.items():
+                file.write(f"      - name: {prop_key}\n")
+                if prop_value.description is not None:
+                    file.write(f"        description: \"{prop_value.description}\"\n")
+                file.write(f"        type: {prop_value.type.type}\n")
+                if prop_value.type.is_array:
+                    file.write("        array: true\n")
+
+
+def export_apigear(directory: Path, root: VSSNode, data_type_tree: VSSNode | None,
+                   layers: typing.Dict[SolutionLayers, Path]):
+    """This method is used to traverse through the root VSS node to build
+       -> ApiGear equivalent string buffer and to serialize it accordingly into a file
+    """
+    module = ApiGearModule()
+    module_name = root.name
+    module_filename = MODULE_FILENAME.format(module_name)
+    module_path = os.path.join(directory, module_filename)
+
+    log.info(f"directory: {directory}")
+    log.info(f"Module name: {module_name}")
+    log.info(f"Module filename: {module_filename}")
+    log.info(f"Module Path: {module_path}")
+
+    if (not layers):
+        layers = {SolutionLayers.CPP: Path("../cppservice")}
```

Comment:
looks fishy to default the location to CWD .. /cppservice.

Reply:
Yeah it should rather be a working directory, changed.